File: HSAPI/src/main/kotlin/com/hobbystock/types/Project.kt
package com.hobbystock.types

import java.util.UUID

data class Project(
        val id: UUID,
        val name: String,
        val description: String?,
        val status: ProjectStatus,
        val startDate: String?,
        val endDate: String?,
        val createdAt: String,
        val updatedAt: String?,
        val items: List<ProjectItem>? = null
)

data class ProjectInput(
        val name: String,
        val description: String?,
        val status: ProjectStatus,
        val startDate: String?,
        val endDate: String?
)

data class ProjectMutationResult(val success: Boolean, val message: String?, val project: Project?)

File: app/src/types/Project.ts
import { ProjectItem } from "./ProjectItem";
import { ProjectSupplyCheck } from "./ProjectSupplyCheck";
import { ProjectStatus } from "./ProjectStatus";

export type Project = {
  id: number;
  name: string;
  description?: string;
  status: ProjectStatus;
  startDate?: string;
  endDate?: string;
  createdAt: string;
  updatedAt?: string;
  items?: ProjectItem[];
  supplyCheck?: ProjectSupplyCheck[];
};

export type ProjectInput = {
  name: string;
  description?: string;
  status: ProjectStatus;
  startDate?: string;
  endDate?: string;
};
